packages/mdx/test/index.test.js

```diff
@@ -296,6 +296,14 @@ it('Should render elements without wrapping blank new lines', async () => {
   expect(result).not.toContain('{`\n`}')
 })
 
+it('Should wrap export in React.memo when memo option is true', async () => {
+  const result = await mdx(`# Test`, {
+    memo: true
```

Comment:
why add an option instead of always doing this?

Reply:
That could work too. I wasn't sure if there was a reason that it isn't being done already, so I figured that an option would be the safe route.

Reply:
The one thing that I'm seeing here, though, is `React.memo` will make the default output coupled to React again. I wonder if we should somehow wrap with `React.memo` in the loader?

Reply:
Oh yeah, I had misinterpreted how the `vue-loader` works and was thinking that it uses a different HAST to JSX function. A few possible solutions that come to mind:

1. Use a `wrapExport` option instead of `memo`.
```js
const result = await mdx(content, {
  wrapExport: "React.memo"
})
```

2. Have `mdx` return separate file parts (either the `MDXContent` function and everything else or something more granular). This would allow more freedom in how the loaders output file contents, and would make the `skipExport` option moot, but also has a larger potential to break things.

```js
const { fn, rest } = await mdx(...);
const code = `/* @jsx mdx */
import React from 'react'
import mdx from '@mdx-js/mdx/create-element'
${rest}

export default ${fn}
`
```

Unless there is compelling reason for `mdx` to return the output broken into parts, I would go with 1 since it is a lot simpler to implement.

Reply:
I like the `wrapExport` idea, especially since it doesn't require breaking the output into multiple parts (I think this could be tricky and unintuitive for users).